app/assets/javascripts/alaveteli_pro/batch_authority_search/results.js

```diff
@@ -66,6 +70,7 @@
 
     $search.on(SearchEvents.domUpdated, update);
     $search.on(SearchEvents.loadingSuccess, updateResults);
+    $search.on(SearchEvents.loadingSuccess, updateDraftId);
```

Comment:
I have a feeling the event you should be listening to is `SearchEvents.rendered` – because that’s the event emitted when `render()` has updated the HTML content of `$results`.

In fact, it makes me wonder, why not just add `$results.find('.js-draft-id').val(DraftBatchSummary.draftId);` in near the end of render(), perhaps on line 54, before the `$search.trigger(SearchEvents.rendered);` ?

Reply:
Thanks, both useful comments.

I quite like the named method for updating the draft ID, so I think:

```js
  var updateDraftId = function updateDraftId() {
    $('.js-draft-id').val(DraftBatchSummary.draftId);
  };

  var render = function render() {
    // …
    if ($results.html() !== content) {
      $results.html(content);
      updateDraftId();
      $search.trigger(SearchEvents.rendered);
    }
  };
```

That look okay?

Reply:
Sure! :+1: